Wikipedia/Code/WMFImageTag+TargetImageWidthURL.m

```diff
@@ -0,0 +1,62 @@
+#import "WMFImageTag+TargetImageWidthURL.h"
+#import "WMFImageURLParsing.h"
+
+@implementation WMFImageTag (TargetImageWidthURL)
+
+/**
+ *  Get URL to image as close to desired target width as possible.
+ *
+ *  @param targetWidth  The desired width for images.
+ *
+ *  @return The URL returned needs to be as close to equal to the targetWidth as possible. Because the image scaler will not scale raster images above their canonical resolution the URL returned here needs to sometimes be modified to just use the canonical URL.
+ */
+- (NSURL*)urlForTargetWidth:(NSUInteger)targetWidth {
+    NSString *tagSrc = self.src;
+    
+    // One of these widths was set for us to even get here.
+    NSNumber *safeCanonicalWidthAssumption = self.dataFileWidth ? self.dataFileWidth : self.width;
+    NSString *canonicalExtension = [[[tagSrc stringByDeletingLastPathComponent] lastPathComponent] pathExtension];
```

Comment:
@montehurd ah, I see. Should this move after  `if (sizeFromSrcUrl != NSNotFound) {` then?